### DISCUSSION TITLE
Using Auth0 Single-Sign-On (SSO) with Filament

### DISCUSSION THREAD

**Discussion by Android63** [filamentphp/filament#10189] [Help]
Is it possible to disable Filament's login and, instead, use Auth0 SSO?

If so, how cam I go about it?

In other Laravel projects I have SSO set up and then put the routes I want to be under auth control inside a middleware group like this:
```
Route::group([
    'middleware' => [
        'auth',
    ],
], function () {
    Route::get('/apply', function () {
        return view('programme-selection');
    })->name('programme-selection');
});
```
My Filament panel is under `/dashboard` but it doesn't have a route, so I'm guessing I can't use the same approach here.

Any pointers please?

**Comment by sakaye** [comment]
Hi @Android63, 

I was able to convert my filament app to use Microsoft SSO. In your PanelProvider class you just need to remove the method calls to login on your main panel. https://filamentphp.com/docs/3.x/panels/users#authentication-features

That panel method also contains the ->authMiddleware([Authenticate::class]) method call which functions the same way as your routes example above. Any pages that load through that panel will use that auth middleware.

**Reply by Android63** [reply]
That's encouraging @sakaye. I removed the login method from my panel class as you did:
```
public function panel(Panel $panel): Panel
{
    return $panel
        ->default()
        ->id('dashboard')
        ->path('dashboard')
//            ->login()
        ->colors([
            'primary' => Color::Blue,
        ])
```
And it directs me to log in via SSO. So far so good.

But after I log in, it tries to open /login and I get a redirect error:
<img width="1358" alt="Screenshot 2023-12-20 at 13 22 53" src="https://github.com/filamentphp/filament/assets/1991226/85e55304-1ee9-49f1-8bb0-0c37b4e34803">

**Reply by sakaye** [reply]
@Android63 You will need to read the docs of the SSO package you are using to see if there is an SSO login route to use.

For example: I'm using this package https://github.com/24Slides/laravel-saml2 and it has built-in saml_route() and saml_url() methods for generating the correct links for login.

In my routes file I use redirect()->saml_route() for my standard /login route.

You will also need to do something similar in your Exceptions/Handler.php for the unauthenticated() method.
